src/modules/FioAddress/components/FioActionSubmit.js

```diff
@@ -163,13 +164,8 @@ class FioActionSubmitComponent extends React.Component<Props, State> {
     const { addressTitles } = this.props
     const { paymentWallet } = this.state
     if (paymentWallet) {
-      const { currencyCode } = paymentWallet.currencyInfo
-      let balance = paymentWallet.balances[currencyCode] ?? '0'
-      if (SPECIAL_CURRENCY_INFO[currencyCode]?.isStakingSupported) {
-        const lockedBalance = paymentWallet.balances[`${currencyCode}${STAKING_BALANCES.locked}`] ?? '0'
-        balance = bns.sub(balance, lockedBalance)
-      }
-      this.setState({ balance: this.formatFio(balance) })
+      const availbleBalance = getAvailableBalance(paymentWallet)
+      this.setState({ balance: this.formatFio(availbleBalance) })
```

Comment:
How does adding an abstracting fix the issue?

Reply:
Oh, this is just to avoid code duplication, the actual fix is in the file above.